org.eclipse.jdt.ls.tests/META-INF/MANIFEST.MF

```diff
@@ -5,8 +5,10 @@ Bundle-Name: %Bundle-Name
 Bundle-SymbolicName: org.eclipse.jdt.ls.tests;singleton:=true
 Bundle-Version: 1.38.0.qualifier
 Export-Package: org.eclipse.jdt.ls.core.internal;x-friends:="org.eclipse.jdt.ls.tests.syntaxserver"
-Bundle-RequiredExecutionEnvironment: JavaSE-17
-Import-Package: org.osgi.framework;version="1.3.0"
+Bundle-RequiredExecutionEnvironment: JavaSE-23
+Import-Package: org.osgi.framework;version="1.3.0",
+ org.eclipse.jdt.internal.javac,
+ org.eclipse.jdt.internal.javac.dom
```

Comment:
Instead of tweaking the MANIFEST.MF, you can consider adding the javac bundle as `extraRequirement` with `target-platform-configuration` block in the pom.xml. This bit of customization could also be encapsulated in the `javac` profile, so it can be easier to merge later.